Public/Src/Cache/ContentStore/Distributed/Utilities/GrpcDistributedPathTransformer.cs

```diff
@@ -16,7 +18,7 @@ namespace BuildXL.Cache.ContentStore.Distributed.Utilities
     /// </summary>
     public class GrpcDistributedPathTransformer : IAbsolutePathTransformer
     {
-        private static readonly string _localMachineName = Environment.MachineName;
+        private static readonly string _localIpAddress = Dns.GetHostEntry(Dns.GetHostName()).AddressList.FirstOrDefault(ip => ip.AddressFamily == System.Net.Sockets.AddressFamily.InterNetwork).ToString();

```

Comment:
I think this code deserves the comment to explain why Environment.MachineName is not good enough here.

Reply:
And this is especially "important" because we have a bunch of other places in the code where we still use Environment.MachineName. So if the new code is the right way to do that I think we should move this logic into a helper and use it everywhere. (or explain why this case is different).

---
In reply to: [274116211](https://github.com/Microsoft/BuildXL/pull/120#discussion_r274116211) [](ancestors = 274116211)

Reply:
Both ways work, so I'll return to MachineName.

---
In reply to: [274116694](https://github.com/Microsoft/BuildXL/pull/120#discussion_r274116694) [](ancestors = 274116694,274116211)